stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -565,7 +593,11 @@ swift::swift_conformsToProtocol(const Metadata *type,
   unsigned failedGeneration = ConformanceCacheGeneration;
 
   // If we didn't have an up-to-date cache entry, scan the conformance records.
+#if defined(_MSC_VER)
+  C.SectionsToScanLock.lock();
+#else
   pthread_mutex_lock(&C.SectionsToScanLock);
```

Comment:
I don't feel good about the proliferation of `#if`s in concurrency code.  The code is complex as is.  Could you continue the work from https://github.com/apple/swift/pull/1731 and move the runtime over to platform-independent mutex wrappers?